qiskit/circuit/controlflow/for_loop.py

```diff
@@ -142,7 +142,7 @@ class ForLoopContext:
         from qiskit import QuantumCircuit
         qc = QuantumCircuit(2, 1)
 
-        with qc.for_loop(None, range(5)) as i:
```

Comment:
This is a bug in doc that is it not being exposed.

Reply:
For the record for others: this is deliberately not exposed documentation - it's an internal private detail of the control-flow builder interface.  I just wrote some documentation in the docstrings of the private components for developers who are modifying the code.  If we build a "developer documentation" / "architecture" section in the future, we could potentially put it there, but for now it's fine just being effectively a code comment.